dotcom-rendering/src/components/Discussion/Comments.tsx

```diff
@@ -214,6 +228,13 @@ export const Comments = ({
 		dispatch({ type: 'addReply', comment });
 	};
 
+	useEffect(() => {
+		if (isWeb && expanded && !loading && enableMobileDiscussionAdsSwitch) {
+			const event = new CustomEvent('comments-loaded');
```

Comment:
Why not reuse the `comments-expanded` custom event and pass in the screen size in the [detail](https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent/detail)? In the commercial codebase, it may be confusing when looking at `comments-expanded-advert.ts` that we have two event listeners `comments-expanded` and `comments-loaded` that are both responding to the same event: the "View more comments" button has been clicked

Reply:
I agree that it isn't ideal that we have two event listeners, but they aren't quite listening for the same event. When the "View more comments" button has been clicked, the comments do not immediately load, so if the commercial code is faster than DCR inserting the comments, there's nothing to insert the ads into and the commercial code fails. It was more robust when testing to wait for the comments to have finished loading instead of listening for the button to be clicked.

The `comments-loaded` event also handles the case where a user changes page, comment number, or changes their comment sort order. For every one of those events, the comments reload and are updated in the DOM. Because of this, we need to rerun the ad insertion code whenever the comments reload. So the `comments-loaded` event is not always linked to the "View more comments" button. Even if we listened for `comments-expanded` initially and added a `setTimeout` to handle the wait for comments to be inserted, we would still need the `comments-loaded` event to handle the page change/filter change case. I hope this makes sense - happy to go through the code in more detail if needed.

Reply:
That makes sense. If the events are different, we should use a different event, as you have implemented. This is much better than using a setTimeout I think.